storage/innobase/handler/ha_innodb.cc

```diff
@@ -19049,10 +19061,10 @@ static MYSQL_SYSVAR_ULONGLONG(buffer_pool_size, innobase_buffer_pool_size,
 static MYSQL_SYSVAR_ULONG(buffer_pool_chunk_size, srv_buf_pool_chunk_unit,
   PLUGIN_VAR_RQCMDARG | PLUGIN_VAR_READONLY,
   "Size of a single memory chunk"
-  " for resizing buffer pool. Online buffer pool resizing happens"
-  " at this granularity. 0 means disable resizing buffer pool.",
+  " for resizing buffer pool. Online buffer pool resizing happens at this"
+  " granularity. 0 means autosize this variable based on buffer pool size.",
   NULL, NULL,
-  128 * 1024 * 1024, 1024 * 1024, LONG_MAX, 1024 * 1024);
+  0, 0, LONG_MAX, 1024 * 1024);
```

Comment:
Please change this to `MYSQL_SYSVAR_SIZE_T`. Is `LONG_MAX` a reasonable maximum size?

Reply:
How important is it for buffer_pool_size `MYSQL_SYSVAR_ULONGLONG` to be the same as the chunk size?

Reply:
MYSQL_SYSVAR_SIZE_T, use the max of SIZE_T. Its capped to the buffer pool size anwyay